derive/examples/std.rs

```diff
@@ -5,32 +5,32 @@ use jsonrpc_derive::rpc;
 #[rpc]
 pub trait Rpc {
 	/// Returns a protocol version
-	#[rpc(name = "protocolVersion")]
+	#[rpc(name = "protocolVersion", returns = "String")]
```

Comment:
While I understand that it's not possible to auto-derive the return types, I'd like to propose two things:
1. Require `returns` attribute only when client is being generated (see my other suggestion to add a parameter to `#[rpc]`)
2. Deduce the return type for types like `Result`, `Future`, etc and allow overriding using `returns`. This will greatly simplify most common cases. I'm fine to have that as a separate PR or just as an open issue.

Reply:
If the return type ends with Result it will take the first generic type argument as the returns type.